scanmem.h

```diff
@@ -29,6 +30,28 @@
 #endif
 */
 
+/* from string.h in glibc */
+#ifndef strdupa
```

Comment:
At the moment no C library uses real functions for this. We could put a check for these into AC_CHECK_FUNCS() but this can't detect macros and I'm not sure if it can detect inline functions. I'd really avoid making this too complex unless we get a real problem somewhere.